tests/plugins/org.polarsys.capella.test.diagram.filters.ju/src/org/polarsys/capella/test/diagram/filters/ju/DiagramTitleBlockFilterTestCase.java

```diff
@@ -0,0 +1,129 @@
+ * first check of this generic test case is to verify that elements that must be filtered by the tested filter are not
+ * filtered firstly.
+ */
+public abstract class DiagramTitleBlockFilterTestCase extends DiagramObjectFilterTestCase {
+
+  protected void getCurrentFilterMappings() {
+    filteredMappingNames.add(IMappingNameConstants.DT_TITLE_BLOCK_CONTAINER);
+    filteredMappingNames.add(IMappingNameConstants.DT_TITLE_BLOCK_LINE_CONTAINER);
+    filteredMappingNames.add(IMappingNameConstants.DT_TITLE_BLOCK_COLUMN_CONTAINER);
+    filteredMappingNames.add(IMappingNameConstants.DT_TITLE_BLOCK_CELL);
+    filteredMappingNames.add(IMappingNameConstants.DT_TITLE_BLOCK_EDGE);
+  }
+  
+  protected void preRunTest() {
+
+    getCurrentDiagram();
+    getCurrentFilterMappings();
+
+    setFilteredObjectMaps();
+
+    for (String id : filteredObjetIDs) {
+      if (!toBeFiltered.values().contains(id)) {
+        fail("Element with id: " + id + " is not available for diagram: " + diagramName + " from project "
+            + projectTestName);
+      }
+    }
+  }
+
+  protected void setFilteredObjectMaps() {
+    for (DDiagramElement elt : diagram.getDiagramElements()) {
+      EObject target = elt.getTarget();
+      if(target != null && target instanceof DAnnotation) {
+        String targetId = ((DAnnotation) target).getUid();
+        String elementMappingName = elt.getMapping().getName();
+        if (filteredObjetIDs.contains(targetId) && filteredMappingNames.contains(elementMappingName)) {
+          toBeFiltered.put(elt, targetId);
+        } else {
+          notToFilter.put(elt, targetId);
+        }
+      }
+    }
+  }
+  
+  /**
+   * Verifies that all elements that should NOT be filtered, are actually NOT filtered by the specified filter
+   */
+  protected void ensurePresenceOfElementsToNotFilter() {
+    Enumeration<DDiagramElement> elementsToNotFilter = notToFilter.keys();
+    while (elementsToNotFilter.hasMoreElements()) {
+      DDiagramElement currentElementToVerify = elementsToNotFilter.nextElement();
+      if (isFilteredByTestedFilter(currentElementToVerify)) {
+        fail("Element: " + ((CapellaElement) currentElementToVerify.getTarget()).getId() + " should not be filtered!");
+      }
+    }
+  }
+
+  /**
+   * Verifies that all elements that should be filtered, are actually filtered by the specified filter
+   */
+  protected void ensureAbsenceOfElementsToFilter() {
+    Enumeration<DDiagramElement> elementsToFilter = toBeFiltered.keys();
+    while (elementsToFilter.hasMoreElements()) {
+      DDiagramElement currentElementToVerify = elementsToFilter.nextElement();
+      if (!isFilteredByTestedFilter(currentElementToVerify)) {
+        fail("Element: " + ((DAnnotation) currentElementToVerify.getTarget()).getUid() + " should be filtered!");
+      }
+    }
+  }
+
+  protected boolean isFilteredByTestedFilter(DDiagramElement elt) {

```

Comment:
rename elt variable

Reply:
I'll keep elt, is a for variable and is not reused later to pose problems that we don't understand what it means.

Reply:
And because in super class is elt.